toffy/qc_metrics_plots.py

```diff
@@ -0,0 +1,72 @@
+from ark.utils.load_utils import load_imgs_from_tree, load_imgs_from_dir
+import matplotlib.pyplot as plt
+import seaborn as sns
+import numpy as np
+from toffy import qc_comp
+from ark.utils import io_utils
+import pandas as pd
+
+def load_normal_fovs(data_dir, 
+	                 normal_tissues,
+                     channels=None,
+                     exclude_channels=None,
+                     img_sub_folder=None):
+    
+    normal_samples = io_utils.list_folders(dir_name=data_dir, substrs=normal_tissues)
+    data = load_imgs_from_tree(data_dir=data_dir, 
+	                    img_sub_folder=img_sub_folder, 
+	                    fovs=normal_samples, 
+	                    channels=channels)    
+    return data
+
+def qc_metrics_plots(data_dir, 
+	                 normal_tissues,
+                     channels=None,
+                     exclude_channels=None,
+	                 qc_metric="99.9th_percentile",
+	                 fig_dir=None):
+    
+    normal_samples = io_utils.list_folders(dir_name=data_dir, substrs=normal_tissues)
+
+    # need to make a nice pandas DF with columns: sample, channel, 99.9th value
+    for i in range(len(normal_tissues)):
+        data = load_normal_fovs(data_dir,
+                                normal_tissues=normal_tissues,
+                                channels=channels)
+        channels = list(data.channels.values)
+        if exclude_channels:
+            channels = [x for x in channels if x not in exclude_channels]
+        
+        
+        # For each tissue type, get all the samples
+        samples = [x for x in normal_samples if normal_tissues[i] in x]
```

Comment:
because i make a different plot for each normal tissue so you either need to load the data during each loop or load the data once and then define the subset of samples matching the tissue for that loop

Reply:
Right now, the difference between `normal_samples`, `samples`, and `normal_tissues` is hard to keep track of. Loading the data necessary for each iteration of the loop only within that loop, rather than subsetting, I think will help reduce the number of variable names getting passed and simplify the logic.